app/models/thread.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+class Thread extends AppModel
+{
+    public $validation = array(
+        'title' => array(
+            'length' => array(
+                'validate_between', 1, 30,
+            ),
+        ),
+    );
+
+    public static function getAll()
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM thread');
+        foreach ($rows as $row) {
+            $threads[] = new Thread($row);
+        }
+        return $threads;
+    }
+
+    public static function get($id)
+    {
+        $db = DB::conn();
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+        return new self($row);
+    }
+
+    public function getComments()
+    {
+        return Comment::getByThreadId($this->id);
+    }
+
+    public function create(Comment $comment)
+    {
+        $this->validate();
+        $comment->validate();
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('invalid thread or comment');
+        }
+        $db = DB::conn();
+        $db->begin();
+        $db->query('INSERT INTO thread SET title = ?, created = NOW()', array($this->title));
+        $this->id = $db->lastInsertId();
+        // write first comment at the same time
+        $this->write($comment);
```

Comment:
this will produce an error since write() method was moved in comment model. please modify to 
```php
$comment->thread_id = $this->id;
$comment->write();